### PR TITLE
Event listeners

### PR DESCRIPTION
**Pull request by fkurmannucsc**
Stopped propagation of typing events to prevent triggering hotkeys.

### CODE REVIEW

**Review by eriq-augustine — CHANGES_REQUESTED** [review]
I'm afk most of the day, so I'll keep it terse.

Instead of putting listeners explicitly on all input fields, just attach them when initializing hotkeys.
Then init hotkeys when you add screenshots (so the new inputs also have the handlers).

This way, we have a fixed one-time addition to the screenshot process and it will handle all future inputs.
Note that your current solution makes it easy for people to mess up in the future (part of writing maintainable code is also making it harder for people to mess up).

**Comment by fkurmannucsc** [comment]
I know I didn't exactly follow your instructions from the comment bit I think the only way to do this is to create a separate function to remove the bubbling from all text inputs i.e. not doing it in the initializeHotkeys() function because that one can only really be called once to prevent duplicate behavior.

I hope I'm not missing something really obvious but this is the best solution I could come up with.

File: static/js/main.js [hunk 1]
```diff
@@ -464,9 +468,16 @@ function initializeHotkeys() {
     });
 }
 
+function removeKeydownSideEffects() {
```
Comment (eriq-augustine) [inline static/js/main.js]: I don't like this function name, it is both too specific and ambiguous.

Too specific: "removeKeydown".
We may choose another key event type (e.g. keyUp).

Ambiguous: "SideEffects".
["Side effect"](https://en.wikipedia.org/wiki/Side_effect_(computer_science)) has a very specific meaning in CS,
and what is happening here is not a CS side effect or even technically a side effect in the looser sense (it is unintentional, but not a side effect).

Mirroring your initializeHotkeys() function, maybe something like "removeHotkeysOnText()".

File: static/js/main.js [hunk 2]
```diff
@@ -464,9 +468,16 @@ function initializeHotkeys() {
...
+    document.querySelectorAll('input[type="text"]').forEach((input) => {
+        input.setAttribute('onkeydown', 'event.stopPropagation()');
+    });
+}
+
 function main() {
     fetchVersion();
     initializeHotkeys();
+    removeKeydownSideEffects();
```
Comment (eriq-augustine) [inline static/js/main.js]: Move this call inside initializeHotkeys().

It's nicer when functions deal with their own fallout.

**Review by eriq-augustine — CHANGES_REQUESTED** [review]
Pretty close.

Response to your comment:
You can remove event listeners before adding them in initializeHotkeys(), thereby making it idempotent.
But what you have here works just fine.

**Review by eriq-augustine — APPROVED** [review]
lgtm

**Pull request merged**